### ISSUE TITLE
Reorg: Unhandled  NotImplementedException with mono 5.2.0 and unity 2017

### ISSUE THREAD

**Issue by kohanis** [AngelDE98/MonoMod#160]
### Description
Unhandled `NotImplementedException` from `AppContext.GetData` in `MonoMod.Switches.TryGetSwitchEnabled` of `MonoMod.Utils` with mono 5.2.0 and unity 2017 mono

**Comment by nike4613** [comment]
Do those builds of Mono target .NET Framework 4.6+?

**Comment by kohanis** [comment]
That's the neat part - mono 5.2.0 targets 4.7) But AppContext.GetData is just
```cs
public static object GetData(string name)
{
	throw new NotImplementedException();
}
```
Unity 2017 mono is targeting 4.6 because it's technically mono 5.0.1, but it has commits between 5.0.1 and 5.2.0 as far as I can tell

**Comment by nike4613** [comment]
Is it only `GetData`? Is `TryGetSwitch` present and implemented?

**Comment by kohanis** [comment]
Yes. Also, as alternative `AppDomain.CurrentDomain.GetData` could be used if `AppContext.GetData` is not present, on frameworks it's literally reroute, idk why mono left `NotImplementedException`
